Fix coding style nits in crypto/asn1/a_strnid.c

InfoHunter opened the pull request:
Since this file is modified in PR #3934, so should be cleaned up
incidentially.

<!--
Thank you for your pull request. Please review these requirements:

Contributors guide: https://github.com/openssl/openssl/blob/master/CONTRIBUTING

Other than that, provide a description above this comment if there isn't one already

If this fixes a github issue, make sure to have a line saying 'Fixes #XXXX' (without quotes) in the commit message.
-->

dot-asm commented on crypto/asn1/a_strnid.c:
```diff
@@ -84,19 +85,21 @@ ASN1_STRING *ASN1_STRING_set_by_NID(ASN1_STRING **out,
+    } else {
         ret =
             ASN1_mbstring_copy(out, in, inlen, inform,
                                DIRSTRING_TYPE & global_mask);
```
Make this two lines instead of three, with `ret = ASN1_mbstring_copy(...`.

InfoHunter commented:
Fixed and new commit added. This time I think you could show your skills to squash : ), but Rich has also fixed the squash script, so we could also try that.

paulidale commented on crypto/asn1/a_strnid.c:
```diff
@@ -144,14 +147,15 @@ ASN1_STRING_TABLE *ASN1_STRING_TABLE_get(int nid)
     }
     tmp = ASN1_STRING_TABLE_get(nid);
-    if (tmp && tmp->flags & STABLE_FLAGS_MALLOC)
+    if (tmp != NULL && tmp->flags & STABLE_FLAGS_MALLOC)
```
I think this would be more readable as:

`if (tmp != NULL && (tmp->flags & STABLE_FLAGS_MALLOC))`

or (better) as:

`if (tmp != NULL && (tmp->flags & STABLE_FLAGS_MALLOC) != 0)`

paulidale replied:
Okay not either way.  All three ways since the original is also good.

paulidale commented:
I can't merge until next week.

richsalz commented:
Squashed and merged and author preserved :)

InfoHunter commented:
> Squashed and merged and author preserved :)

didn't find the commit in master

InfoHunter commented:
> didn't find the commit in master

Oops, find now...seems your script works perfectly : )
